File: src/taeeun/week21/빗물.cpp
#include <iostream>
#include<algorithm>
#include <vector>

using namespace std;
int main() {
	int W, H;
	int rain = 0;
	cin >> H >> W;
	vector <int> block(W);
	for (int i = 0; i < W; i++) {
		cin >> block[i];
	}
	for (int i = 1; i < W - 1; i++) {
		// i는 현재 열
		int leftMax = 0, rightMax = 0;
		//왼쪽 열 중의 최대 값
		for (int j = 0; j < i; j++) {
			leftMax = max(block[j], leftMax);
		}
		//오른쪽 열 중의 최대 값
		for (int j = i + 1; j < W; j++) {
			rightMax = max(block[j], rightMax);
		}
		if (min(leftMax, rightMax)-block[i]>0) rain+= min(leftMax, rightMax) - block[i];
	}
	cout << rain;

}

File: src/geonhyeong/week21/빗물.py
## 빗물
## Question Link: https://www.acmicpc.net/problem/14719

import sys
input = sys.stdin.readline

h, w = map(int, input().split())
block = list(map(int, input().split()))

res = 0
for i in range(1, w - 1): # 첫 번째 칸과 마지막칸은 물이 고일수 없음
    leftMax = max(block[:i]) # 기준에서 가장 높은 왼쪽
    rightMax = max(block[i+1:]) # 기준에서 가장 높은 오른쪾

    compare = min(leftMax, rightMax)

    if block[i] < compare:
        res += compare - block[i]

print(res)
